includes/admin/import/class-batch-import-payments.php

```diff
@@ -447,39 +454,46 @@ private function set_customer( $row ) {
 			}
 
 			// Make sure we found a customer. Create one if not.
-			if( empty( $customer->id ) ) {
+			if ( empty( $customer->id ) ) {
 
 				if ( ! $customer instanceof EDD_Customer ) {
-					$customer = new EDD_Customer;
+					$customer = new EDD_Customer();
 				}
+			}
+
+			$name = '';
+
+			if ( ! empty( $this->field_mapping['name'] ) && ! empty( $row[ $this->field_mapping['name'] ] ) ) {
 
+				$name = sanitize_text_field( $row[ $this->field_mapping['name'] ] );
+
+			} else {
 				$first_name = '';
 				$last_name  = '';
-
-				if( ! empty( $this->field_mapping['first_name'] ) && ! empty( $row[ $this->field_mapping['first_name'] ] ) ) {
+				if ( ! empty( $this->field_mapping['first_name'] ) && ! empty( $row[ $this->field_mapping['first_name'] ] ) ) {
 
 					$first_name = sanitize_text_field( $row[ $this->field_mapping['first_name'] ] );
 
 				}
 
-				if( ! empty( $this->field_mapping['last_name'] ) && ! empty( $row[ $this->field_mapping['last_name'] ] ) ) {
+				if ( ! empty( $this->field_mapping['last_name'] ) && ! empty( $row[ $this->field_mapping['last_name'] ] ) ) {
 
 					$last_name = sanitize_text_field( $row[ $this->field_mapping['last_name'] ] );
 
 				}
-
-				$customer->create( array(
-					'name'  => $first_name . ' ' . $last_name,
-					'email' => $email
-				) );
-
-				if( ! empty( $canonical_id ) && (int) $canonical_id !== (int) $customer->id ) {
-					$customer->update_meta( '_canonical_import_id', $canonical_id );
-				}
-
+				$name = $first_name . ' ' . $last_name;
 			}
 
+			$customer->create(
+				array(
+					'name'  => $name,
+					'email' => $email,
```

Comment:
It looks like `$email` could be not set. (This may have already been the case before your changes, but let's fix it anyway.)